### DISCUSSION TITLE
shortcuts for standard highlighting tasks

### DISCUSSION THREAD

**Discussion by mjmay-kinneret** [WinEdt-Team/WinEdt-Support#20] [3.1 Lounge]
Hello WinEdt,
I create many assignments and exams using WinEdt. I end up using **textbf**, _emph_, and <ins>underline</ins> environments a lot and find it frustrating that there are no predefined keyboard shortcuts for them.  I can add keyboard shortcuts for bold and italics using the options menu, but underline doesn't seem to even have a macro file defined for it.

Is there a way to add an underline macro to the base WinEdt package so that adding keyboard shortcuts for it will be easier?   Or at least send me a sample macro file that I can put in my local definitions?

Thanks
michael

**Comment by urmel1960** [comment]
You can just copy an existing macro and modify accordingly - like this:

````
// -*- ASCII:EDT -*-

BeginGroup;
IfSel('0','=',>
      'SelWord(1);>
       IfSel(''0'',''='',>
             ''Ins("\underline{}");>
               CharLeft;'',>
             ''InsLabel("\underline","{","}")'');',>
      'InsLabel("\underline","{","}");');
EndGroup;
End;
````

Then you can add another menu entry into the MainMenu.ini using options interface (don't forget  to load the ini file after editing - check also if the shortcut is suitable for your setup). I put it for me into Insert/Font

````
ITEM="Underline"
      CAPTION="&Underline"
      IMAGE="Underline"
      MACRO="Exe('%b\Exec\Underline.edt');"
      REQ_DOCUMENT=1
      SHORTCUT="16460::Ctrl+L"                         
````

**Reply by Alex-Simonic** [reply]
Underline (without a keyboard shortcut) has now been added to the default menu and drop-down toolbar button (for the upcoming WinEdt 11.2).
